iam-persistence/src/main/java/it/infn/mw/iam/persistence/repository/IamAupSignatureRepositoryImpl.java

```diff
@@ -68,7 +68,9 @@ private void deleteSignature(IamAccount account) {
   @Override
   public IamAupSignature createSignatureForAccount(IamAup aup, IamAccount account,
       Date currentTime) {
-
+    if (account.isServiceAccount()) {
+      throw new IamAupSignatureUpdateError(account, "As user '%s' is a service account, AUP signature operation not allowed");
```

Comment:
To avoid repetition, I'd define the message directly in the IamAupSignatureUpdateError constructor

Reply:
Done